### PR TITLE
Adding grid isOnBorder() tests and deleting duplicate method

### PR DESCRIPTION
**Pull request by Aayush-Agnihotri**
### Description

Replace this with description
image

### CODE REVIEW

File: engine/grid.py [hunk 1]
```diff
@@ -161,31 +161,6 @@ def determine_active_waypoints(self, node):
 
     # --------------------- METHODS TO FINISH INITIALIZATION OF ACTIVATED GRID -------------- #
 
-    def is_on_border(self, row, col, row_limit, col_limit):
```
Comment (sazhang02) [inline engine/grid.py:164]: Nice! 👍

File: tests/compilation_tests/grid_test.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 from engine.grid import Grid
 from engine.kinematics import get_vincenty_x, get_vincenty_y
 from engine.mission import ControlMode
+from tests.functionality_tests.test_activation_shapes import is_on_border
```
Comment (sazhang02) [inline tests/compilation_tests/grid_test.py:7]: Since you removed the duplicate is_on_border, I think we can delete this import statement and use the remaining method from engine.grid (which is already imported)

**Review by sazhang02 — CHANGES_REQUESTED** [review]
FANTASTIC job Aayush on your first PR! 🎉 🎉 🎉  One very small change below and it should be good to merge afterward 🎉 

Some additional PR suggestions for future reference (a bit on the longer side for explanation purposes!):
- Github has a cool feature called Draft PRs (I recently discovered this like last sem), which allows you to make a PR but indicates it's still WIP (while PR means ready to merge to main) and you can turn it into a PR later. Think it would be a neat way to get suggestions from fellow teammates when you're not completely done. I believe when you have the option to make a PR, there's a lil drop-down arrow next to it to make it a draft instead if u need to make on in the future)
- In addition to the nice code cov image, I would add what terminal commands you ran to get that for whoever's reviewing to see so they can reproduce

File: engine/grid.py (hunk 1 above)
Comment (AdityaKompella6) [reply engine/grid.py:164]: I love this deletion exquisite!

**Review by AdityaKompella6 — APPROVED** [review]
Beaueitiful

**Review by sazhang02 — APPROVED** [review]
Nice!!! Let's merge this :DDDD
